signatures/helpers/helpers.go

```diff
@@ -435,3 +437,9 @@ func GetProtoHTTPByName(
 
 	return trace.ProtoHTTP{}, fmt.Errorf("protocol HTTP: type error (should be trace.ProtoHTTP, is %T)", arg.Value)
 }
+
+func CloneProperties(m *detect.SignatureMetadata) {
+	// do a shallow clone of Properties map getting a new reference
+	// avoiding leaking the original pointer
```

Comment:
```suggestion
// CloneMetadataProperties makes a shallow clone of the Metadata
// Properties map, ensuring that there is always a new reference to it.
//
// TODO: since this helper is a workaround to avoid data races,
// perhaps a better solution would be to convert Properties into
// a concrete structure.
func CloneMetadataProperties(m *detect.SignatureMetadata) {
```

WDYT of an outer comment and rename the helper?

Reply:
Its better! tks!